tests/clvm/coin_store.py

```diff
@@ -118,30 +118,15 @@ def update_coin_store_for_spend_bundle(
         err = self.validate_spend_bundle(spend_bundle, now, max_cost)
         if err != 0:
             raise BadSpendBundleError(f"validation failure {err}")
-        for new_coin in spend_bundle.additions():
+        additions = spend_bundle.additions()
+        removals = spend_bundle.removals()
+        for new_coin in additions:
             self._add_coin_entry(new_coin, now)
-        for spent_coin in spend_bundle.removals():
+        for spent_coin in removals:
             coin_name = spent_coin.name()
-            try:
```

Comment:
What are these changes?

Reply:
I believe this is an accidental reversal of a change that was made in singleton-driver and merged into the pools branch to handle ephemeral coins.  If @richardkiss confirms that the `tests/clvm/test_singletons.py` test passes with these changes then I think it doesn't matter.

Reply:
To be clear, although this is called `CoinStore`, it's really a testing-only `CoinStore` drop-in replacement that should be renamed something else. It didn't handle ephemeral coins properly prior to this change. I believe the `try/except` block removed here may have conjured missing spends out of nothing and so may have been flawed, although I'm not 100% sure. We could probably do with more testing of this testing structure.

Currently, our contract development cycle is ill-defined and the testing suite is unbearably slow. Replacements like this will likely be more useful in future for rapid test cycles in developing new contracts.

Reply:
I think the try/catch was flawed, but it made the tests pass.  If `tests/clvm/test_singletons.py` passes without the try/catch block then it's fine to remove, otherwise it should probably be added back in.

Reply:
The `tests/clvm/test_singletons.py` tests work for me.